### PR TITLE
Update rubocop-performance gemspec

### PR DESCRIPTION
**Pull request by Kuraiou**
# What

* use 1.12.x for rubocop-performance.

# Why

It's the most recent version and bugfix versions, generally speaking, should be allowable (i.e. use `~>` instead of `=`).

### CODE REVIEW

File: standard.gemspec
Comment (searls) [inline standard.gemspec:24]: We have made an intentional decision to lock to specific releases of rubocop and rubocop-performance, in order that standard's behavior at any given version is reproducible in all cases for a given target Ruby. If you'd like to bump to the latest rubocop-performance release, that's great (we just hadn't gotten to it yet), but please lock to the specific version.

File: config/base.yml
Comment (searls) [inline config/base.yml:927]: This cop does not support Rubies older than 2.3 (based on [this issue linked from the rubocop PR](https://github.com/rails/rails/pull/43502)), so to enable it in the base means it should be disabled in the 2.2 config file.

File: lib/standard/version.rb
Comment (searls) [inline lib/standard/version.rb:2]: Generally, it's considered bad form to bump the version number of a library in a PR, as it's the maintainer's responsibility to manage what goes into each release, in addition to version control, packaging, and publishing of each release (which, if you see the git log of the `version.rb` file, you'll see we're very particular about). As a result, it can sometimes come across as pushy or forward to bump the version of a library in a PR (the implication being an imperative "cut a release as soon as this is merged just for this change")

**Review by searls — CHANGES_REQUESTED** [review]

**Comment by camilopayan** [comment]
Should be covered in the latest v1.5.0 version.

**Pull request closed without merging**
